### PR TITLE
Task/DES-1158 - $q.all Requests in projects area and pass browser data between states

### PR DESCRIPTION
**Pull request by Strmiska**
This seems like a big change, but it is really just the same change being made to multiple component/template combinations for each project type.

1. Rewrote the `$onInit` sections in the following areas:
- Working Directory - project-view.component.js
- Curation Directory - project-directory.component.js
- Publication Preview - (all project types)
- Curation Pipeline Selection - (all project types)

2. Pass browser data between states and reload entity listings only when necessary

3. Removed `project-data` as it is no longer being used

4. Modified `files-listing` to accept a params binding for hiding/showing features

5. Fixed an issue in the pipeline causing entity selections to be overwritten when editing

### CODE REVIEW

**Comment by jarosenb** [comment]
While you're refactoring stuff here, it might be cool to replace the underscore calls with ES6 array methods.

**Comment by Strmiska** [comment]
Working on getting FR set up with this...

File: designsafe/static/scripts/data-depot/components/projects/pipeline-selection/pipeline-selection-sim.component.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import PipelineSelectionSimTemplate from './pipeline-selection-sim.component.html';
 import _ from 'underscore';
```
Comment (Strmiska) [inline designsafe/static/scripts/data-depot/components/projects/pipeline-selection/pipeline-selection-sim.component.js]: remove import underscore

File: designsafe/static/scripts/data-depot/components/projects/project-view/project-view.component.html [hunk 2]
```diff
@@ -107,5 +107,13 @@
...
-    <ui-view class="project-data"></ui-view>
+  <div ng-if="$ctrl.loading">
+      <h3 class="text-center">
+          <i class="fa fa-spinner fa-spin"></i> Loading...
+      </h3>
+  </div>
+  <div ng-if="!$ctrl.loading">
+    <files-listing browser="$ctrl.browser"></files-listing>
+  </div>
+    <!-- <ui-view class="project-data"></ui-view> -->
```
Comment (Strmiska) [inline designsafe/static/scripts/data-depot/components/projects/project-view/project-view.component.html]: delete commented code

File: designsafe/static/scripts/data-depot/components/projects/publication-preview/publication-preview-field-recon.component.js [hunk 3]
```diff
@@ -18,62 +18,55 @@ class PublicationPreviewFieldReconCtrl {
     }
```
Comment (Strmiska) [inline designsafe/static/scripts/data-depot/components/projects/publication-preview/publication-preview-field-recon.component.js]: remove _ import

File: designsafe/static/scripts/data-depot/components/projects/publication-preview/publication-preview-hyb-sim.component.js [hunk 4]
```diff
@@ -18,62 +18,55 @@ class PublicationPreviewHybSimCtrl {
     }
```
Comment (Strmiska) [inline designsafe/static/scripts/data-depot/components/projects/publication-preview/publication-preview-hyb-sim.component.js]: remove _ underscore

File: designsafe/static/scripts/data-depot/components/projects/publication-preview/publication-preview-sim.component.js [hunk 5]
```diff
@@ -18,62 +18,55 @@ class PublicationPreviewSimCtrl {
     }
```
Comment (Strmiska) [inline designsafe/static/scripts/data-depot/components/projects/publication-preview/publication-preview-sim.component.js]: remove underscore import

File: designsafe/static/scripts/data-depot/components/projects/publication-preview/publication-preview.component.js [hunk 6]
```diff
@@ -22,65 +22,55 @@ class PublicationPreviewCtrl {
         this.readOnly = this.$state.current.name.indexOf('publishedData') === 0;
```
Comment (Strmiska) [inline designsafe/static/scripts/data-depot/components/projects/publication-preview/publication-preview.component.js]: remove underscore import

**Pull request merged**